### ISSUE TITLE
Declare "float", "int" in code and configure columns as "decimal" and "mediumint" in MySQL

### ISSUE THREAD

**Issue by vijayrajasekaran** [balderdashy/sails-mysql#199]
Hello,

I see that it is not possible to declare decimal and mediumint in models (Sails.js), instead i declared them as float and int. In MySQL as decimal and mediumint. And made sure, the migrate option is set as safe so that it doesn't mess the database tables. It is working fine.

Do i need to worry about this since i am also using "tinyint" instead of "int", i believe these things will help save space and decimal can be used for currency fields. Is there a workaround?

Thanks.

**Comment by albertkim** [comment]
Quick q, is there any active development being done on this module? The last commits were from a while back, and there is room for improvement here...

**Comment by dmarcelino** [comment]
Hi @albertkim, recently several new collaborators (me included) have been added to the waterline project (and its modules and adapters). Without making any promises, development on this adapter should become more active now. Feel free to submit PRs and open issues and the collaborators will look and assess them.

**Comment by devinivy** [comment]
@vijayrajasekaran it seems like you have been careful enough and understand the implications for migrations.  So you should be alright!  If you have any concrete ideas how this adapter could more flexibly cater to these scenarios, let us know with a feature request.

**Comment by vijayrajasekaran** [comment]
@devinivy Thank you so much for your response. I will go ahead and use it this way till this module get support for Decimal, Tinyint, Mediumint etc.

**Comment by devinivy** [comment]
:+1:

**Comment by devinivy** [comment]
Reopening marked as a feature request.

**Comment by dmarcelino** [comment]
There are 2 parts to this:
1. update anchor to accept "decimal" and "mediumint";
2. update sails-mysql to create the proper column types.

**Comment by devinivy** [comment]
I wonder if we should use the `size` attribute property to specify the size of the `integer` type.  That makes the options more portable, and non-mysql specific.  Different adapters can use the `size` property however they like while maintaining a standard waterline type (integer).

**Comment by dmarcelino** [comment]
Indeed @devinivy that sounds like a good generic solution for this problem. I don't think anchor is doing anything with `size`, it probably just passes through to the adapter. So, in case of an integer, the adapter would interpret this a the number of bytes? Apart from changes in sails-mysql we would just need to update the documentation. I like it! Sounds like a good candidate for waterline-docs `0.11` and sails-mysql `0.11`.

**Comment by vijayrajasekaran** [comment]
I tried to declare the column as 'STRING' instead of 'FLOAT' in Sails.js Models and then as Decimal in MySql database, it is working as accepted. Now, i am confused should i go with 'FLOAT' or 'STRING' in Models?

**Comment by superlbr** [comment]
Wating to use decimal??

**Comment by vijayrajasekaran** [comment]
I just want it to be displayed as (4,2) with exact value like 01.50, 10.00, 00.10,

**Comment by tjwebb** [comment]
@superlbr sails.js and its adapters are built and maintained by the community of developers who use it. If you'd like to collaborate with some of the other users here and submit a PR with this feature, it would be greatly appreciated.

**Comment by sailsbot** [comment]
Thanks for posting, @vijayrajasekaran.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
